### PR TITLE
Ignore exceptions when closing stream

### PR DESCRIPTION
**Pull request by alenkacz**
JIRA issues: DCOS-54077

Co-authored-by: Tarun Gupta Akirala <takirala@users.noreply.github.com>

### CODE REVIEW

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/HttpClient.scala [hunk 1]
```diff
@@ -43,7 +44,8 @@ object HttpClient {
     fetchResponse(uri, DEFAULT_RETRIES, headers: _*)
       .flatMap { case (responseData, conn) =>
         Future(processResponse(responseData))
-          .ensure(responseData.contentStream.close())
+          // IOException when closing stream should not fail the whole request, that's why we're ignoring exceptions with Try
+          .ensure(Try(responseData.contentStream.close()))
           .ensure(conn.disconnect())
```
Comment (takirala) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/HttpClient.scala]: wdyt of wrapping disconnect call also in a Try ?

Comment (takirala) [reply cosmos-server/src/main/scala/com/mesosphere/cosmos/HttpClient.scala]: And i think the CI is failing cuz `ensure` is expecting a unit value.

Comment (alenkacz) [reply cosmos-server/src/main/scala/com/mesosphere/cosmos/HttpClient.scala]: I checked and it does not advertise to be throwing any exceptions... they actually even try catch IO exception inside the disconnect call. So I found it not necessary. But if you want to be super-safe, happy to do it

Comment (takirala) [reply cosmos-server/src/main/scala/com/mesosphere/cosmos/HttpClient.scala]: ah okay. we don't need Try in that case.

File: cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/handler/ResourceProxyHandlerIntegrationSpec.scala [hunk 2]
```diff
@@ -36,6 +37,34 @@ final class ResourceProxyHandlerIntegrationSpec extends FreeSpec
       assertResult(HttpResponseStatus.FORBIDDEN)(response.error.status)
     }
 
+    "fail when the url is present but is in-accessible" in {
+      /**
```
Comment (takirala) [inline cosmos-integration-tests/src/main/scala/com/mesosphere/cosmos/handler/ResourceProxyHandlerIntegrationSpec.scala]: @alenkacz i don't like this test very much but this is the best compromise i have found so far. lmk if you have any thoughts.

**Pull request merged**